### PR TITLE
Move pretrain into package

### PR DESCRIPTION
**Pull request by carmocca**
Part of #982

### CODE REVIEW

File: lit_gpt/pretrain.py
Comment (awaelchli) [inline lit_gpt/pretrain.py:1]: Will all scripts be moved to the lit_gpt folder at the top level? Or should there maybe be a directory to separate it from the rest of lit-gpt?

Importing it in `lit_gpt.__init__` should still enable it to work as `python -m lit_gpt.pretrain`

File: lit_gpt/pretrain.py [hunk 1]
```diff
@@ -1,13 +1,7 @@
 # Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
 
-"""
-This script is adapted from TinyLlama:
-https://github.com/jzhang38/TinyLlama/blob/main/pretrain/tinyllama.py
-"""
```
Comment (awaelchli) [inline lit_gpt/pretrain.py:6]: While the script has changed a lot, there are still some artifacts left over, for example the weight initialization. So maybe keep this?

Comment (carmocca) [reply lit_gpt/pretrain.py:6]: What if we reference what hasn't changed? I can link tinyllama in the weight initialization function

File: lit_gpt/pretrain.py
Comment (carmocca) [reply lit_gpt/pretrain.py:1]: I don't have a strong preference here. Happy to follow whatever you or @rasbt prefer

**Review by awaelchli — APPROVED** [review]

**Pull request merged**